library/core/src/ptr/alignment.rs

```diff
@@ -391,7 +395,9 @@ mod verify {
     impl kani::Arbitrary for Alignment {
         fn any() -> Self {
             let align = kani::any_where(|a: &usize| a.is_power_of_two());
```

Comment:
This should create an arbitrary usize and a Kani assume is safe

Reply:
Ok, so something like this?

```rust
let align: usize = kani::any();
let ret = unsafe { mem::transmute::<usize, Alignment>(align) };
kani::assume(ret.is_safe());
```

I didn't do this originally because I thought that transmuting an _arbitrary_ usize to an Alignment would cause UB (specifically, this case):

> First and foremost, creating an instance of any type with an invalid state is going to cause arbitrary chaos that can't really be predicted. Do not transmute 3 to bool. Even if you never do anything with the bool. Just don't.

Alignments only take AlignmentEnums, which are usizes that are powers of two. We do restrict the range of `align` with the `kani::assume` later, but we've already done the transmute on the unrestricted value to get `ret`--does that not cause UB?

We could implement `kani::Arbitrary` for `AlignmentEnum`, e.g.:

```rust
impl kani::Arbitrary for AlignmentEnum {
  let val: usize = kani::any_where(|a: &usize| a.is_power_of_two());
  // this transmute is safe because we've restricted `val` to powers of two,
  // so we'll always produce a valid variant of AlignmentEnum
  unsafe { mem::transmute::<usize, AlignmentEnum>(val) }
}

impl kani::Arbitrary for Alignment {
  let align: AlignmentEnum = kani::any();
  let alignment = unsafe {mem::transmute::<AlignmentEnum, Alignment>(align) };
  kani::assume(alignment.is_safe());
  alignment
}
```

Reply:
No... I want less unsafe code. :) From what I understand, the safety and validity invariant of Alignment type is the same. So a derive should be enough.

Reply:
I really don't think this should be using transmute. Can we implement `Arbitrary` for `Alignment` and using it here instead?

Reply:
I'm not sure what you mean--the transmute is happening inside the implementation of `Arbitrary` for `Alignment`. I applied your suggestion about the derive macros to get rid of the transmute; LMK if that's what you had in mind.